ledger/src/blockstore.rs

```diff
@@ -3937,20 +3933,18 @@ impl Blockstore {
     /// Returns a mapping from each elements of `slots` to a list of the
     /// element's children slots.
     pub fn get_slots_since(&self, slots: &[Slot]) -> Result<HashMap<Slot, Vec<Slot>>> {
-        let slot_metas: Result<Vec<Option<SlotMeta>>> = self
-            .meta_cf
-            .multi_get(slots.iter().copied())
-            .into_iter()
-            .collect();
-        let slot_metas = slot_metas?;
-
-        let result: HashMap<Slot, Vec<Slot>> = slots
-            .iter()
-            .zip(slot_metas)
-            .filter_map(|(slot, meta)| meta.map(|meta| (*slot, meta.next_slots.to_vec())))
-            .collect();
+        let keys = self.meta_cf.multi_get_keys(slots.iter().copied());
+        let slot_metas = self.meta_cf.multi_get(&keys);
+
+        let mut slots_since: HashMap<Slot, Vec<Slot>> = HashMap::with_capacity(slots.len());
+        for meta in slot_metas.into_iter() {
+            let meta = meta?;
+            if let Some(meta) = meta {
+                slots_since.insert(meta.slot, meta.next_slots);
+            }
+        }
```

Comment:
To match existing behavior:
- Do bubble up a rocksdb error
- Do not error if we don't have a `meta` for any of the given slots
    - FWIW, this should never be the case since we figure out what slots to pass into this function by looking at currently frozen banks in `ReplayStage`
    
Lastly, we don't need the meta's after, so use `into_iter()` and take over `meta.next_slots` instead of `meta.next_slots.to_vec()`